### PR TITLE
Close file objects, requests.Responses

### PR DESCRIPTION
**Pull request by jku**
**Fixes issue #1099**:

maybe also #1123 (but I've only gone through the code exercised by test_updater.py and test_download.py)

**Description of the changes being introduced by the pull request**:

Try to close all file objects and requests.Responses that are left open during  test_updater.py and test_download.py execution.

**Please verify and check that the pull request fulfills the following
requirements**:

- [x] The code follows the [Code Style Guidelines](https://github.com/secure-systems-lab/code-style-guidelines#code-style-guidelines)
- [ ] Tests have been added for the bug fix or new feature
- [ ] Docs have been added for the bug fix or new feature

### CODE REVIEW

**Comment by jku** [comment]
These are good changes I think, but I guess there's also a case to be made for silencing ResourceWarnings about these things in the testsuite:
* the same warnings don't appear in production 
* as far as I understand, cpython can handle automatic closing just fine. The issues would only appear on other interpreters (pypy?)

I'm not in favor of silencing: just mentioning it

File: tuf/download.py [hunk 1]
```diff
@@ -251,24 +251,21 @@ def _download_file(url, required_length, STRICT_REQUIRED_LENGTH=True):
     # Always stream to control how requests are downloaded:
     # http://docs.python-requests.org/en/master/user/advanced/#body-content-workflow
```
Comment (MVrachev) [inline tuf/download.py]: Can you update or remove the link? It leads to a 404 webpage.

File: tuf/download.py [hunk 2]
```diff
@@ -251,24 +251,21 @@ def _download_file(url, required_length, STRICT_REQUIRED_LENGTH=True):
...
     #
-    # We will always manually close Responses, so no need for a context
-    # manager.
-    #
     # Always set the timeout. This timeout value is interpreted by requests as:
     #  - connect timeout (max delay before first byte is received)
     #  - read (gap) timeout (max delay between bytes received)
     # These are NOT overall/total, wall-clock timeouts for any single read.
     # http://docs.python-requests.org/en/master/user/advanced/#timeouts
```
Comment (MVrachev) [inline tuf/download.py]: Can you update or remove the link? It leads to a 404 webpage.

**Review by MVrachev — COMMENTED** [review]
Great changes!
I tried them locally and I don't see `Resource warnings` when running `tests_download.py` and `tests_updater.py`.

Can you increase the scope of this pr and fix the other instances in the other tests files  `Resource warnings` (as mentioned in #1123) or that will be much of work right now?

**Comment by jku** [comment]
> Can you increase the scope of this pr and fix the other instances in the other tests files `Resource warnings` (as mentioned in #1123) or that will be much of work right now?

This is the scope for now I think. Finding/fixing the trickier ones can take quite a bit of time

File: tuf/client/updater.py [hunk 3]
```diff
@@ -3281,11 +3283,6 @@ def download_target(self, target, destination_directory,
     trusted_length = target['fileinfo']['length']
     trusted_hashes = target['fileinfo']['hashes']
 
-    # '_get_target_file()' checks every mirror and returns the first target
-    # that passes verification.
-    target_file_object = self._get_target_file(target_filepath, trusted_length,
-        trusted_hashes, prefix_filename_with_hash)
-
     # We acquired a target file object from a mirror.  Move the file into place
```
Comment (joshuagl) [inline tuf/client/updater.py]: Minor nit: This comment is now out of date, we haven't actually acquired the target file yet.

File: tuf/download.py [hunk 4]
```diff
@@ -382,16 +379,8 @@ def _download_fixed_amount_of_data(response, temp_file, required_length):
...
-    # Whatever happens, make sure that we always close the connection.
-    response.close()
     raise tuf.exceptions.SlowRetrievalError(str(e))
 
-  except:
-    # Whatever happens, make sure that we always close the connection.
-    response.close()
-    raise
-
-  response.close()
```
Comment (joshuagl) [inline tuf/download.py:394]: Nice. This should probably have been a `try/except/finally` to ensure the response was always closed.

**Review by joshuagl — APPROVED** [review]
Thanks for these changes, it's nice to see the number of warnings produced by the tests being reduced.

File: tuf/download.py (hunk 4 above)
Comment (jku) [reply tuf/download.py:394]: Yes, that was definitely a potential future bug

**Comment by jku** [comment]
The last change only updated the code comment in the first commit based on Joshuas comment. No other changes.

**Pull request merged**